### ISSUE TITLE
quiet and verbose_ping cannot ping continuously

### ISSUE THREAD

**Issue by EclectickMedia** [l4m3rx/python-ping#14]
# Preamble

The current iteration of the functions `quiet_ping` and `verbose_ping` do not allow for continuous ping requesting due to their current `for i in range(x)` structure.

The current iteration simply counts down from the kwarg `count`, which no matter how theoretically large of a number we supply, will never achieve the necessary functionality. 

Even if we were to provide a number at the maximum of integer size, it would present a few unnecessary behaviours:
1. An unnecessarily large number inherently takes up more memory than it is worth.
2. We could even need to go so far as to determine wether to provide a 32bit max or a 64bit maximum integer.

# Solution Number 1

We could potentially solve the problem by simply providing a generator that continuously spits out numbers for `range` to consume.
- I feel that this could present more complex code than is necessary to achieve the functionality we are looking for. Consider the following code as an example generator object:
```python
def gen():
    i = 0

    while 1:  # Continuously loop
        i += 1
        yield i
```
- Unless we reset the number, this would still result in extremely large numbers and run into complication **1**.
  - Reseting the number sequence could potentially cause strange behaviour in `range`.

# Solution Number 2

We could use an `if else` block to determine whether to use the current `for i in range(x)` behaviour or use a `while` loop to continuously request pings.
-This could run into issues meshing with other frameworks due to the nature of the while loop not relinquishing control.

# Solution Number 3 - *A More Elegant Solution*

Converting the `quiet` and `verbose_ping` functions to generators would allow us to use a `for` loop to infinitely generate ping statistics.

This can be achieved by placing the functions in a `while 1` loop with an exit clause checking for max iterations completion, and a yield after every successful call to `do_one`.

The calling code/function could then call the function in the following manner:
```python
for results in verbose_ping():
    pass
```

# Summary *(TL;DR)*

The current implementation of `quiet` and `verbose_ping` does not allow truelly endless pinging (for those who really like to collect statistics), and there are a couple ways we could get that done.

I am not sure which way to proceed, but will likely go with *Solution Number 3*

--*Ari*

**Comment by EclectickMedia** [comment]
I should add, I have no problem accomplishing all of these, but I just need some input.

**Comment by l4m3rx** [comment]
Go for it :)

**Comment by EclectickMedia** [comment]
This is closed via #18
